include/nbl/video/IGPUBuffer.h

```diff
@@ -28,7 +28,21 @@ class NBL_API IGPUBuffer : public asset::IBuffer, public IDeviceMemoryBacked, pu
 		// Vulkan: const VkBuffer*
 		virtual const void* getNativeHandle() const = 0;
 
-		struct SCreationParams : asset::IBuffer::SCreationParams, IDeviceMemoryBacked::SCreationParams {};
+		struct SCreationParams : asset::IBuffer::SCreationParams, IDeviceMemoryBacked::SCreationParams
+		{
+SCreationParams() = default;
+			
+SCreationParams(const asset::IBuffer::SCreationParams& params) 
+	: asset::IBuffer::SCreationParams(params)
+{ }
+
+SCreationParams& operator =(const asset::IBuffer::SCreationParams& rhs)
+{
+	static_cast<asset::IBuffer::SCreationParams&>(*this) = rhs;
+	return *this;
+}
```

Comment:
why is the indentation so messed up?

Reply:
I fixed it here
https://github.com/Devsh-Graphics-Programming/Nabla/commit/02343aadd5a0bba2ae034fbeb6e4929ac2c57c42